homeassistant/components/lg_thinq/number.py

```diff
@@ -0,0 +1,206 @@
+"""Support for number entities."""
+
+from __future__ import annotations
+
+import logging
+
+from thinqconnect import DeviceType
+from thinqconnect.devices.const import Property as ThinQProperty
+from thinqconnect.integration import ActiveMode, TimerProperty
+
+from homeassistant.components.number import (
+    NumberDeviceClass,
+    NumberEntity,
+    NumberEntityDescription,
+    NumberMode,
+)
+from homeassistant.const import PERCENTAGE, UnitOfTemperature, UnitOfTime
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from . import ThinqConfigEntry
+from .entity import ThinQEntity
+
+NUMBER_DESC: dict[ThinQProperty, NumberEntityDescription] = {
+    ThinQProperty.FAN_SPEED: NumberEntityDescription(
+        key=ThinQProperty.FAN_SPEED,
+        translation_key=ThinQProperty.FAN_SPEED,
+    ),
+    ThinQProperty.LAMP_BRIGHTNESS: NumberEntityDescription(
+        key=ThinQProperty.LAMP_BRIGHTNESS,
+        translation_key=ThinQProperty.LAMP_BRIGHTNESS,
+    ),
+    ThinQProperty.LIGHT_STATUS: NumberEntityDescription(
+        key=ThinQProperty.LIGHT_STATUS,
+        native_unit_of_measurement=PERCENTAGE,
+        translation_key=ThinQProperty.LIGHT_STATUS,
+    ),
+    ThinQProperty.TARGET_HUMIDITY: NumberEntityDescription(
+        key=ThinQProperty.TARGET_HUMIDITY,
+        device_class=NumberDeviceClass.HUMIDITY,
+        native_unit_of_measurement=PERCENTAGE,
+    ),
+    ThinQProperty.TARGET_TEMPERATURE: NumberEntityDescription(
+        key=ThinQProperty.TARGET_TEMPERATURE,
+        native_unit_of_measurement=UnitOfTemperature.CELSIUS,
+        translation_key=ThinQProperty.TARGET_TEMPERATURE,
+    ),
+    ThinQProperty.WIND_TEMPERATURE: NumberEntityDescription(
+        key=ThinQProperty.WIND_TEMPERATURE,
+        native_unit_of_measurement=UnitOfTemperature.CELSIUS,
+        translation_key=ThinQProperty.WIND_TEMPERATURE,
+    ),
```

Comment:
Should we add a device class?

Reply:
This same as temperature in sensor.
We don't use device class to support Fahrenheit dynamically.
ex> When a user change UoM in the mobile app, the cloud send the temperature to the integration with UoM.
Keep it as is.

Reply:
The device_class converts the Celsius's min/max when config's unit is Fahrenheit. But native_step is not converted by config's different unit so the same issue as async_set_native_value occurs.
- climate: https://github.com/home-assistant/core/pull/125663#discussion_r1765990233

We don't want to use devices_class(NumberDeviceClass.TEMPERATURE) except for climate.